magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merge_stereo.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+import joblib
+import numpy as np
+import yaml
+from magicctapipe import __version__
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import (
+    rc_lines,
+    slurm_lines,
+)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def MergeStereo(target_dir, env_name, source, NSB_match, cluster):
+    """
+    This function creates the bash scripts to run merge_hdf_files.py in all DL2 subruns.
+
+    Parameters
+    ----------
+    target_dir: str
+        Path to the working directory
+    """
+
+    process_name = source
+    if NSB_match:
+        stereo_DL1_dir = f"{target_dir}/v{__version__}/{source}"
+    else:
+        stereo_DL1_dir = f"{target_dir}/v{__version__}/{source}/DL1/Observations"
+
+    listOfNightsLST = np.sort(glob.glob(f"{stereo_DL1_dir}/DL1Stereo/*"))
+    if cluster == 'SLURM':
+        for nightLST in listOfNightsLST:
+            stereoMergeDir = (
+                f"{stereo_DL1_dir}/DL1Stereo/{nightLST.split('/')[-1]}/Merged"
+            )
+            os.makedirs(f"{stereoMergeDir}/logs", exist_ok=True)
+            if not os.listdir(f"{nightLST}"):
+                continue
+            if len(os.listdir(nightLST)) < 3:
```

Comment:
why 3? is it only subruns or also other files?

Reply:
In the DL1 stereo directory there will always be a `logs` folder and a `Merged` one even in case the Dl1 stereo production failed. But if there is not at least one file too (i.e., 3 elements), we cannot/ should not merge things. But this is still to be tested and could maybe be changed using wildcards on the file extensions

Reply:
basing this on some other files that might be in the same directory is a bit dangerous. how about using
```len(glob.glob("f{nightLST}/dl1_stereo*.h5")```